service/Microsoft.Azure.Devices/JobClient/JobClient.cs

```diff
@@ -25,6 +25,26 @@ public static JobClient CreateFromConnectionString(string connectionString)
             return new HttpJobClient(iotHubConnectionString);
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
+        /// </summary>
+        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
+        protected virtual void Dispose( bool disposing )
+        {
+            if (disposing)
+            {
+                // Call CloseAsync() synchronously avoiding AggregateException.
+                this.CloseAsync().GetAwaiter().GetResult();
+            }
+        }
+
```

Comment:
Remove this implementation. Let the derived  concrete implementation deal with this.